### PR TITLE
Issue 10 snippet helper

### PR DESCRIPTION
**Pull request by m-l-1**
#10

### CODE REVIEW

File: R/snippets.R [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+#' 
+#' @param force Force the re-installation when the snippets are already installed.
+#' 
+#' @import snippr
+#' @import dplyr
+#' @importFrom fs path_home_r path_ext_set
+#' @importFrom  cli cat_bullet
+#' @importFrom  purrr keep discard map
+#' @export
+add_tm_snippets <- function(force = FALSE){
```
Comment (xni7) [inline R/snippets.R:14]: For users without ~/.R/snippets/r.snippet this reports an error

> tidymodules::add_tm_snippets()
Error in file(con, "r") : cannot open the connection
In addition: Warning message:
In file(con, "r") :
  cannot open file '~/.R/snippets/r.snippets': No such file or directory

**Comment by DivadNojnarg** [comment]
Started to test:
- `add_module`works well

Maybe add a slot for nested modules as well as:

```r
# module R6Class ---------------------------------

#' R6 Class representing the module
#'
#' @description
#' UI and server code necessary for the module
#' @details
#' More details
#'
#' @export
module <- R6::R6Class(
  "module",
  inherit = tidymodules::TidyModule,
  public = list(
    #' @field nested_mod nested modules. This is updated by the initialize function.
    nested_mod = list(),
    
    #' @description
    #' Module initialization
    #' @param ... NULL by default.
    #' @return A new `module` object.
    initialize = function(...) {
      super$initialize(...)
      
      # initialize nested modules
      # self$nested_mod[[1]] <- Nested$new(inherit = TRUE)
      
      # define ports
      self$definePort({
    
      })
      
    },
    
    #' @description
    #' Create the module UI
    #' @return ...
    ui = function() {
      tagList(
        # UI elements go here
      )
    },
    
    #' @description
    #' Server function description
    #' @param input Shiny input object.
    #' @param output Shiny output object.
    #' @param session Shiny session object.
    #' @return The server function.
    server = function(input, output, session) {
      
      super$server(input, output, session)
      
      # call all nasted modules below
      # self$nested_mod[[1]]$callModule()
      
      # update output ports
      self$assignPort({
        # self$updateOutputPort(...)
      })
      
    }
  )
)
```

**Comment by DivadNojnarg** [comment]
Second test fails:

`tidymodules::add_tm_snippets()`:

```
Error in file(con, "r") : cannot open the connection
In addition: Warning message:
In file(con, "r") :
cannot open file '/home/rstudio-user/.R/snippets/r.snippets': No such file or directory
```

File: DESCRIPTION [hunk 2]
```diff
@@ -19,7 +19,8 @@ Imports:
     digest,
     R6,
     visNetwork,
-    methods
+    methods,
+    snippr
```
Comment (xni7) [inline DESCRIPTION]: {snippr} failed to install in the latest version of 🔱. It's only on Github?Suggest further testing on DaVinci before releasing.

File: inst/rstudio/r.snippets [hunk 3]
```diff
@@ -0,0 +1,70 @@
+snippet tm.mod.new
```
Comment (xni7) [inline inst/rstudio/r.snippets:1]: snippets work in RSP1.2, but not in RStudio 1.3.776 on Mac

File: R/add_module.R [hunk 4]
```diff
@@ -0,0 +1,323 @@
...
+#' @importFrom cli cat_bullet
+#' @importFrom utils file.edit
+#' @importFrom fs path_abs path file_create
+#' @importFrom snippr snippets_read
+#' 
+#' @export
+add_module <- function(
+  name,
+  inherit = "TidyModule",
+  pkg = getwd(),
```
Comment (xni7) [inline R/add_module.R]: Does this function support only R package structure only? what about a regular shiny app? Probably need to document in the vignette on the good practice of organizing {tm} class files

File: inst/rstudio/r.snippets (hunk 3 above)
Comment (DivadNojnarg) [reply inst/rstudio/r.snippets:1]: RStudio 1.3.776 is a preview version

File: R/snippets.R (hunk 1 above)
Comment (m-l-1) [reply R/snippets.R:14]: It should be solved now.

File: DESCRIPTION (hunk 2 above)
Comment (m-l-1) [reply DESCRIPTION]: I just installed it!

File: inst/rstudio/r.snippets (hunk 3 above)
Comment (m-l-1) [reply inst/rstudio/r.snippets:1]: We should be fine with our version, will check that one later.

**Review by m-l-1 — COMMENTED** [review]
Made a few changes addressing most of your feedbacks, could you please re-check?

File: R/add_module.R (hunk 4 above)
Comment (m-l-1) [reply R/add_module.R]: Here is my proposition (which is actually implemented in the function)
- the default path is always current wd => option `path`
- if R sub-dir exists, add it
- class name always starts with capital letter => option `name`
- there is a default prefix added to filename "tm_" => option `prefix`

Comment (m-l-1) [reply R/add_module.R]: The function supports both !

**Review by DivadNojnarg — APPROVED** [review]

**Review by xni7 — APPROVED** [review]
Great job! certainly will help newcomers

File: R/snippets.R (hunk 1 above)
Comment (bzkrouse) [reply R/snippets.R:14]: Works great if the file does not exist.  However, I am getting a different error if the directories ~/.R or ~/.R/snippets do not exist (those seem to be created only after the user edits snippets for the first time):

Error: [ENOENT] Failed to open '~/.R/snippets/r.snippets': no such file or directory

Comment (m-l-1) [reply R/snippets.R:14]: Well spotted! Is it better now?

Comment (bzkrouse) [reply R/snippets.R:14]: Yes, working great! Thanks for the fix.

Comment (m-l-1) [reply R/snippets.R:14]: @bzkrouse  Could you approve ?-)

**Review by bzkrouse — APPROVED** [review]
@m-l-1 Really nice additions to the package! These helpers are user-friendly and should definitely help reduce overhead when creating new classes.  Looking forward to using them.

**Pull request merged**